tensorflow_quantum/python/util.py

```diff
@@ -344,6 +345,86 @@ def _symbols_in_op(op):
         "tfq.util.get_supported_gates().")
 
 
+def is_expression_approx_eq(exp_1, exp_2, atol):
+    """Compare possibly symbolic expressions for approximate equality.
+
+    Coefficient based approximate equality.  If no symbol is present in
+    `exp_1` or `exp_2`, then return true if the expressions are approximately
+    equal.  If the expressions contain symbols, return true if the two symbols
+    are the same and their coefficients are approximately equal.
+    """
+    if not isinstance(atol, numbers.Real):
+        raise TypeError("atol must be a real number.")
+    s_1 = serializer._symbol_extractor(exp_1)
+    s_2 = serializer._symbol_extractor(exp_2)
+    v_1 = serializer._scalar_extractor(exp_1)
+    v_2 = serializer._scalar_extractor(exp_2)
+    if isinstance(s_1, numbers.Real) and isinstance(s_2, numbers.Real):
+        s_eq = cirq.approx_eq(s_1, s_2, atol=atol)
+    elif isinstance(s_1, sympy.Symbol) and isinstance(s_2, sympy.Symbol):
+        s_eq = str(s_1) == str(s_2)
+    else:
+        s_eq = False
+    v_eq = cirq.approx_eq(v_1, v_2, atol=atol)
+    return s_eq and v_eq
+
+
+def is_gate_approx_eq(gate_true, gate_deser, atol=1e-5):
+    """Compares gates in the allowed TFQ gate set.
+
+    Serialized gates can only have simple expressions of the form s*n or n*s,
+    where s is a `sympy.Symbol` and n is some number.  Checks that each
+    corresponding expression in `gate_true` and `gate_deser` are approximately
+    equal.
```

Comment:
Might want to rephrase here since serialization doesn't really impact the definition of our approximate equality. It might be simpler to just say something like: "Gates in TFQ support symbols, numbers or a single product of a real number and a symbol as their parameters. This function behaves like `cirq.approx_eq` specialized for these kinds of gates so that TFQ can support approximate equality in gates containing symbols: <snippet from your earlier comment in this PR> where as cirq does not." or something.

Reply:
Done